documentation/ZENKO-2982_xdm_gui_create_a_storage_location

wabernat opened the pull request:
moved/added Zenko create a storage location docs to XDM UI section.

This adds documentation for creating a storage location using the XDM GUI per ZENKO-2982.

wabernat committed:
documentation/ZENKO-2982_xdm_gui_create_a_storage_location

moved/added Zenko create a storage location docs to XDM UI section.

bert-e commented:
# Hello wabernatscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @rahulreddy

* @tmacro

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+      special |product|-access user to ensure access to the Metadata service and the
+      ability to perform operations on the bucket:
+
+      ::
```
I would add the json key here for syntax highlighting. I think its like `..codeblock:: json` or something.

wabernat replied:
done

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+      * Scality RING with sproxyd Connector
+      * Scality XCore
+      * Wasabi
+      * A Zenko local filesystem
```
Use product for Zenko here?

wabernat replied:
I don't know. Neither does Product. This is one case where I've backed away from the full-on sed -i 's/Zenko/XDM/g' approach: I want to verify that the filesystem is actually capable of being renamed "XDM" easily before I assume that we can change all references to it with a wave of the marketing wand.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+because it is a |product|-native filesystem, it shares authentication and related
+credentialing tasks, which are addressed elsewhere in the |product| UI.
+
+For more information, see :ref:`Zenko Local`.
```
Zenko here as well but it's a ref so this one might be intentional.

wabernat replied:
yeah, in this case, I can't make a reference to a variable, so we'll have to sort this and other references out by hand and hard-code once we get a clear directive on what the product is going to be called.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The RING maintains stability and redundancy in its object data stores by way of
+a bootstrap list. To access a RING directly using sproxyd, you must enter at
```
when we reference our other products or services ( maybe tools too?) should we add a reference link to it some how? Not sure where exactly but it might be nice.

wabernat replied:
This requires intersphinx implementation. @tcarmet has been working with me to get intersphinx operational on docs-ring-s3c. When I get that sorted out, I'll take it to XDMko.

miniscruff replied:
ok @tcarmet  let me know if you need help with intersphinx, I have used it before with decent success.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+ingest out-of-band updates from NFS mounts. AWS metadata is ingested in an
+initial setup, then changes are mapped via a regularly scheduled cron
+job. |product| develops its own namespace for the Amazon bucket and can perform
+metadata-\ related tasks (CRR, metadata search, lifecycle management, etc.) on
```
any reason why this dash word is doing `one\- two` compared to all the other ones just being `one-two`?

wabernat replied:
That was a line break/continuation that got shifted off the line end. 

Good spot. Fixed.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+      special |product|-access user to ensure access to the Metadata service and the
+      ability to perform operations on the bucket:
+
+      ::
```
add json key for syntax highlighting, I think it's `..codeblock:: json`

wabernat replied:
I am experiencing déja vu...

...and I fixed it.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+      * Scality RING with sproxyd Connector
+      * Scality XCore
+      * Wasabi
+      * A Zenko local filesystem
```
Zenko here instead of `|product|`

wabernat replied:
as discussed above.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+
+      Adding a location requires credentials (an access key and a secret key).
+      Though nothing prevents you from using account-level credentials when
+      Zenko requests credentials for a location, it is a best practice to enter
```
another zenko

wabernat replied:
Done.

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_storage_location.rst:
```diff
@@ -0,0 +1,278 @@
+because it is a |product|-native filesystem, it shares authentication and related
+credentialing tasks, which are addressed elsewhere in the |product| UI.
+
+For more information, see :ref:`Zenko Local`.
```
another zenko but this one is a ref, so that might be intentional

miniscruff commented on docs/docsource/operation/XDM_GUI/location_management/adding_a_transient_source_storage_location.rst:
```diff
@@ -0,0 +1,60 @@
+.. _Adding a Transient Source Storage Location:
```
this file and the next file do not use `|product|` for zenko references, not sure if intentional.

wabernat replied:
This file is basically a place-holder so that the cross-references hold up. I will revisit this file in future tickets (either to edit it or delete it).

miniscruff reviewed (commented):
Mostly small things.

@tcarmet not sure if git-lfs is enabled but we might want to consider it for all the images being added. Granted they do not change much.

wabernat committed:
implemented requested fixes as appropriate.

bert-e commented:
# Conflict




There is a conflict between your branch `documentation/ZENKO-2982_xdm_gui_create_a_storage_location` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`documentation/ZENKO-2982_xdm_gui_create_a_storage_location`).

```bash
 $ git fetch
 $ git checkout origin/documentation/ZENKO-2982_xdm_gui_create_a_storage_location
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:documentation/ZENKO-2982_xdm_gui_create_a_storage_location
```

wabernat committed:
Merge branch 'development/2.0' into documentation/ZENKO-2982_xdm_gui_create_a_storage_location

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @rahulreddy

* @tmacro

wabernat commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @rahulreddy

* @tmacro










*The following options are set:* **approve**

wabernat committed:
Merge branch 'development/2.0' into documentation/ZENKO-2982_xdm_gui_create_a_storage_location

wabernat committed:
documentation/ZENKO-2982_xdm_gui_create_a_storage_location

Added "add a storage location" section to XDM. Updated links. Deleted similar
Orbit section.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`



Please check the status of the associated issue ZENKO-2982.

Goodbye wabernatscality.
